### ISSUE TITLE
Paket hangs forever if using symlinks

### ISSUE THREAD

**Issue by incca-konstantin-savchenko** [fsprojects/Paket#3411]
### Description
I have following folder structure:

	├── Project1
	│    └── ...
	│    └── symlink1
	│    └── symlink2
	│    └── P1.csproj
	│    └── paket.references
	├── Project2
	│    └── ...
	│    └── symlink1
	│    └── symlink2
	│    └── symlink3
	│    └── P2.csproj
	│    └── paket.references
	└──  (...)
	└──  MySolution.sln
	└──  paket.dependencies

Our solution has multiple projects and there are symlinks inside project folders. The symlinks are mapped to local network storage. They are not included in any project and only provide access to shared resources. Some of them have considerable amount of data (20-50GB).

Executing paket (install/update/restore) in this configuration takes eternity.
After:
`.paket/paket restore -v`
Is get: "Parsing (...)\paket.dependencies" and nothing happens for hours. :/

### Known workarounds
After removing all symlinks paket runs as expected and takes only 10-20 seconds.

Is there any possibility to exclude certain paths from processing?
(Paket version 5.186.0)
Thanks

**Comment by forki** [comment]
Oh wow.

Are 5he symlinks only in folders that have a references file in it? If so
we could possibly come up with something in the references file.

Am Do., 8. Nov. 2018, 08:51 hat incca-konstantin-savchenko <
notifications@github.com> geschrieben:

**Comment by incca-konstantin-savchenko** [comment]
Yes, more or less. There are some more symlinks but it seems only those "heavy-weight" linked folders cause the problem.

**Comment by forki** [comment]
Currently I won't be able to do this myself, but I think we would accept a PR that creates a solution to ignore folders and files from paket's csproj / references file search. I think we should create a `.paketignore` file with similar features like `.gitignore`. For the start it's probably enough to just allow ignore patterns.
